diff -Nur Amaya.bak/amaya/AHTBridge.c Amaya/amaya/AHTBridge.c
--- Amaya.bak/amaya/AHTBridge.c	2004-04-14 14:29:31.000000000 +0000
+++ Amaya/amaya/AHTBridge.c	2005-10-21 14:59:41.000000000 +0000
@@ -179,7 +179,7 @@
    int 		       socket;
    ms_t                now = HTGetTimeInMillis();
    
-   socket = (SOCKET) data;
+   socket = (SOCKET) ((intptr_t) data);
    v = HASH (socket);
 
    /* convert the FD into an HTEventType which will allow us to find the
diff -Nur Amaya.bak/amaya/AHTURLTools.c Amaya/amaya/AHTURLTools.c
--- Amaya.bak/amaya/AHTURLTools.c	2004-01-29 16:06:43.000000000 +0000
+++ Amaya/amaya/AHTURLTools.c	2005-10-21 14:59:41.000000000 +0000
@@ -1219,7 +1219,8 @@
   ----------------------------------------------------------------------*/
 void ExtractTarget (char *aName, char *target)
 {
-   int    lg, i;
+   int    lg;
+   intptr_t	i;
    char  *ptr;
    char  *oldptr;
 
@@ -1241,7 +1242,7 @@
 	  }
 	while (ptr);
 
-	i = (int) (oldptr) - (int) (aName);	/* name length */
+	i = (intptr_t) (oldptr) - (intptr_t) (aName);	/* name length */
 	if (i > 1)
 	  {
 	     aName[i - 1] = EOS;
diff -Nur Amaya.bak/amaya/EDITimage.c Amaya/amaya/EDITimage.c
--- Amaya.bak/amaya/EDITimage.c	2004-04-14 14:29:31.000000000 +0000
+++ Amaya/amaya/EDITimage.c	2005-10-21 14:55:46.000000000 +0000
@@ -91,11 +91,11 @@
   char               tempfile[MAX_LENGTH];
   char               tempname[MAX_LENGTH];
   int                i, c1, cN;
-  int                val;
+  intptr_t           val;
   ThotBool           change, isHTML;
 
   isHTML = FALSE;
-  val = (int) data;
+  val = (intptr_t) data;
   switch (ref - BaseImage)
     {
     case FormAlt:
diff -Nur Amaya.bak/amaya/EDITstyle.c Amaya/amaya/EDITstyle.c
--- Amaya.bak/amaya/EDITstyle.c	2004-04-14 14:29:31.000000000 +0000
+++ Amaya/amaya/EDITstyle.c	2005-10-21 14:59:51.000000000 +0000
@@ -1735,9 +1735,9 @@
   ----------------------------------------------------------------------*/
 void StyleCallbackDialogue (int ref, int typedata, char  *data)
 {
-  int               val;
+  intptr_t               val;
 
-  val = (int) data;
+  val = (intptr_t) data;
 #ifdef DEBUG_CLASS_INTERF
   if (typedata == INTEGER_DATA)
     fprintf (stderr, "StyleCallbackDialogue(%d,%d) \n", ref, (int) data);
diff -Nur Amaya.bak/amaya/HTMLbook.c Amaya/amaya/HTMLbook.c
--- Amaya.bak/amaya/HTMLbook.c	2004-02-25 10:06:13.000000000 +0000
+++ Amaya/amaya/HTMLbook.c	2005-10-21 14:59:51.000000000 +0000
@@ -545,9 +545,9 @@
   ----------------------------------------------------------------------*/
 void CallbackPrint (int ref, int typedata, char *data)
 {
-  int                 val;
+  intptr_t                 val;
 
-  val = (int) data;
+  val = (intptr_t) data;
   switch (ref - BasePrint)
     {
     case FormPrint:
diff -Nur Amaya.bak/amaya/MENUconf.c Amaya/amaya/MENUconf.c
--- Amaya.bak/amaya/MENUconf.c	2004-04-14 14:29:31.000000000 +0000
+++ Amaya/amaya/MENUconf.c	2005-10-21 15:05:07.000000000 +0000
@@ -1126,7 +1126,7 @@
   ----------------------------------------------------------------------*/
 static void CacheCallbackDialog (int ref, int typedata, char *data)
 {
-  int                 val;
+  intptr_t            val;
 
   if (ref == -1)
     {
@@ -1136,7 +1136,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - CacheBase)
 	{
 	case CacheMenu:
@@ -1439,7 +1439,7 @@
   ----------------------------------------------------------------------*/
 static void ProxyCallbackDialog (int ref, int typedata, char *data)
 {
-  int                 val;
+  intptr_t            val;
 
   if (ref == -1)
     {
@@ -1449,7 +1449,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - ProxyBase)
 	{
 	case ProxyMenu:
@@ -2150,7 +2150,7 @@
   ----------------------------------------------------------------------*/
 static void GeneralCallbackDialog (int ref, int typedata, char *data)
 {
-  int                 val;
+  intptr_t            val;
 
   if (ref == -1)
     {
@@ -2160,7 +2160,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - GeneralBase)
 	{
 	case GeneralMenu:
@@ -2618,7 +2618,7 @@
   ----------------------------------------------------------------------*/
 static void PublishCallbackDialog (int ref, int typedata, char *data)
 {
-  int                 val;
+  intptr_t            val;
 
   if (ref == -1)
     {
@@ -2628,7 +2628,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - PublishBase)
 	{
 	case PublishMenu:
@@ -3040,7 +3040,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - BrowseBase)
 	{
 	case BrowseMenu:
@@ -3357,7 +3357,7 @@
   ----------------------------------------------------------------------*/
 static void ColorCallbackDialog (int ref, int typedata, char *data)
 {
-  int val;
+  intptr_t val;
 
   if (ref == -1)
     {
@@ -3367,7 +3367,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - ColorBase)
 	{
 	case ColorMenu:
@@ -3700,7 +3700,7 @@
   ----------------------------------------------------------------------*/
 static void GeometryCallbackDialog (int ref, int typedata, char *data)
 {
-  int val;
+  intptr_t val;
 
   TtaDestroyDialogue (GeometryBase + GeometryMenu);
   if (ref == -1)
@@ -3711,7 +3711,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - GeometryBase)
 	{
 	case GeometryMenu:
@@ -3903,7 +3903,7 @@
   ----------------------------------------------------------------------*/
 static void LanNegCallbackDialog (int ref, int typedata, char *data)
 {
-  int val;
+  intptr_t val;
 
   if (ref == -1)
     {
@@ -3913,7 +3913,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - LanNegBase)
 	{
 	case LanNegMenu:
@@ -4215,7 +4215,7 @@
   ----------------------------------------------------------------------*/
 static void ProfileCallbackDialog (int ref, int typedata, char *data)
 {
-  int   val;
+  intptr_t   val;
   char *ptr;
 
   if (ref == -1)
@@ -4224,7 +4224,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - ProfileBase)
 	{
 	case ProfileMenu:
@@ -4443,7 +4443,7 @@
   ----------------------------------------------------------------------*/
 static void TemplatesCallbackDialog (int ref, int typedata, char *data)
 {
-  int val;
+  intptr_t val;
 
   if (ref == -1)
     {
@@ -4453,7 +4453,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - TemplatesBase)
 	{
 	case TemplatesMenu:
@@ -4720,7 +4720,7 @@
   ----------------------------------------------------------------------*/
 static void AnnotCallbackDialog (int ref, int typedata, char *data)
 {
-  int                 val;
+  intptr_t                 val;
 
   if (ref == -1)
     {
@@ -4730,7 +4730,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - AnnotBase)
 	{
 	case AnnotMenu:
diff -Nur Amaya.bak/amaya/Mathedit.c Amaya/amaya/Mathedit.c
--- Amaya.bak/amaya/Mathedit.c	2004-04-14 14:29:31.000000000 +0000
+++ Amaya/amaya/Mathedit.c	2005-10-21 14:59:51.000000000 +0000
@@ -1474,7 +1474,7 @@
 static void CallbackMaths (int ref, int typedata, char *data)
 {
   Document           doc;
-  int                val = (int) data;
+  intptr_t           val = (intptr_t) data;
 
   ref -= MathsDialogue;
   if (ref == MenuMaths1)
@@ -1520,7 +1520,7 @@
 			 gpointer   data )
 {
   InitMaths = FALSE;
-  TtaDestroyDialogue ((int) data);
+  TtaDestroyDialogue ((intptr_t) data);
   return TRUE;
 }
 #endif /* _GTK */
diff -Nur Amaya.bak/amaya/SVGedit.c Amaya/amaya/SVGedit.c
--- Amaya.bak/amaya/SVGedit.c	2004-02-25 10:06:14.000000000 +0000
+++ Amaya/amaya/SVGedit.c	2005-10-21 14:59:51.000000000 +0000
@@ -2046,7 +2046,7 @@
 static void CallbackGraph (int ref, int typedata, char *data)
 {
   Document           doc;
-  int                val = (int) data;
+  intptr_t           val = (intptr_t) data;
 
   ref -= GraphDialogue;
   if (ref == MenuGraph1)
@@ -2086,7 +2086,7 @@
 			 gpointer   data )
 {
   PaletteDisplayed = FALSE;
-  TtaDestroyDialogue ((int) data);
+  TtaDestroyDialogue ((intptr_t) data);
   return TRUE;
 }
 #endif /* _GTK */
diff -Nur Amaya.bak/amaya/UIcss.c Amaya/amaya/UIcss.c
--- Amaya.bak/amaya/UIcss.c	2004-04-27 13:16:23.000000000 +0000
+++ Amaya/amaya/UIcss.c	2005-10-21 14:59:51.000000000 +0000
@@ -628,11 +628,12 @@
   Element         firstSel, lastSel;
   char           *ptr = NULL, *localname = NULL;
   int             j, firstChar, lastChar;
-  int             val, sty;
+  intptr_t        val;
+  int             sty;
   CSSCategory     category;
   ThotBool        found;
 
-  val = (int) data;
+  val = (intptr_t) data;
   category = CSS_Unknown;
   sty = 0; /* document style order */
   switch (ref - BaseCSS)
diff -Nur Amaya.bak/amaya/init.c Amaya/amaya/init.c
--- Amaya.bak/amaya/init.c	2004-04-27 13:16:23.000000000 +0000
+++ Amaya/amaya/init.c	2005-10-21 15:01:33.000000000 +0000
@@ -639,7 +639,7 @@
   ----------------------------------------------------------------------*/
 void ExtractParameters (char *aName, char *parameters)
 {
-   int            lg, i;
+   intptr_t      lg, i;
    char          *ptr;
    char          *oldptr;
 
@@ -661,7 +661,7 @@
 	  }
 	while (ptr);
 
-	i = (int) (oldptr) - (int) (aName);	/* name length */
+	i = (intptr_t) (oldptr) - (intptr_t) (aName);	/* name length */
 	if (i > 1)
 	  {
 	     aName[i - 1] = EOS;
@@ -5500,7 +5500,7 @@
     sep = URL_SEP;
   else
     sep = DIR_SEP;
-  val = (int) data;
+  val = (intptr_t) data;
   if (ref - BaseDialog == OptionMenu)
     /* a popup menu corresponding to a SELECT element */
     ReturnOption = val;
diff -Nur Amaya.bak/amaya/libmanag.c Amaya/amaya/libmanag.c
--- Amaya.bak/amaya/libmanag.c	2004-04-14 14:29:31.000000000 +0000
+++ Amaya/amaya/libmanag.c	2005-10-21 14:59:51.000000000 +0000
@@ -603,7 +603,7 @@
 void CallbackLibrary (int ref, int typedata, char *data)
 {
 #ifdef _SVG
-  int                 val;
+  intptr_t            val;
   Document            svgDoc, res, libDoc;
   char               *id, *buffer;
   char                tempname[MAX_LENGTH], pathname[MAX_LENGTH];
@@ -612,7 +612,7 @@
   Element             copiedElement, el = NULL;
   PRule               PRuleSearch;
 
-  val = (int) data + 1;
+  val = (intptr_t) data + 1;
   switch (ref - BaseLibrary)
     {
     case FormLibrary:
diff -Nur Amaya.bak/amaya/trans.c Amaya/amaya/trans.c
--- Amaya.bak/amaya/trans.c	2004-04-27 13:16:24.000000000 +0000
+++ Amaya/amaya/trans.c	2005-10-21 14:59:51.000000000 +0000
@@ -2363,9 +2363,9 @@
 void TransCallbackDialog (int ref, int typedata, char* data)
 {
   strTransDesc        *trans = NULL;
-  int                  val;
+  intptr_t             val;
 
-  val = (int) data;
+  val = (intptr_t) data;
   switch (ref - TransBaseDialog)
     {
     case TransMenu:
diff -Nur Amaya.bak/annotlib/ANNOTmenu.c Amaya/annotlib/ANNOTmenu.c
--- Amaya.bak/annotlib/ANNOTmenu.c	2004-01-09 12:24:11.000000000 +0000
+++ Amaya/annotlib/ANNOTmenu.c	2005-10-21 14:50:54.000000000 +0000
@@ -97,7 +97,7 @@
 
 #endif /* __STDC__ */
 {
-  int                 val;
+  intptr_t                 val;
 
   if (ref == -1)
     {
@@ -107,7 +107,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - CustomQueryBase)
 	{
 	case CustomQueryMenu:
@@ -868,7 +868,7 @@
 char             *data;
 #endif /* __STDC__ */
 {
-  int val;
+  intptr_t val;
 
   if (ref == -1)
     {
@@ -878,7 +878,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - AnnotFilterBase)
 	{
 	case AnnotFilterMenu:
diff -Nur Amaya.bak/thotlib/base/fileaccess.c Amaya/thotlib/base/fileaccess.c
--- Amaya.bak/thotlib/base/fileaccess.c	2005-10-21 14:12:06.000000000 +0000
+++ Amaya/thotlib/base/fileaccess.c	2005-10-21 14:16:33.000000000 +0000
@@ -584,7 +584,7 @@
        while (ptr != NULL);
 
        /* the length of the directory part */
-       i = ((int) (oldptr) - (int) (text)) / sizeof (char);
+       i = ((intptr_t) (oldptr) - (intptr_t) (text)) / sizeof (char);
        if (i > 1)
 	 {
 	   strncpy (aDirectory, text, i);
diff -Nur Amaya.bak/thotlib/base/structlist.c Amaya/thotlib/base/structlist.c
--- Amaya.bak/thotlib/base/structlist.c	2004-04-27 13:16:24.000000000 +0000
+++ Amaya/thotlib/base/structlist.c	2005-10-21 14:20:16.000000000 +0000
@@ -509,7 +509,7 @@
 	  fprintf (fileDescriptor, "%s", pRe1->SrOrigName);
 	  /* ecrit le nom du schema de structure de l'element */
 	  fprintf (fileDescriptor, "(%s %x)", pNode->ElStructSchema->SsName,
-		   (unsigned int)pNode->ElStructSchema);
+		   (intptr_t)pNode->ElStructSchema);
 	}
       /* ecrit l'URI associee au schema de structure */
       if (pNode->ElStructSchema->SsUriName == NULL)
diff -Nur Amaya.bak/thotlib/dialogue/appdialogue.c Amaya/thotlib/dialogue/appdialogue.c
--- Amaya.bak/thotlib/dialogue/appdialogue.c	2004-04-27 13:16:24.000000000 +0000
+++ Amaya/thotlib/dialogue/appdialogue.c	2005-10-21 15:19:54.000000000 +0000
@@ -2291,7 +2291,7 @@
 {
   GdkEventKey *event;
   GtkWidget *entry;
-  int frame;  
+  intptr_t frame;  
   char *text;
   
   if (all_event->type == GDK_KEY_PRESS)
@@ -2299,7 +2299,7 @@
       event = (GdkEventKey *) all_event;      
       if (event->keyval == GDK_Escape)
 	{
-	  frame = (int) user_data;
+	  frame = (intptr_t) user_data;
 	  entry = (GtkWidget *) gtk_object_get_data (GTK_OBJECT (w), 
 						     "entry");
 	  RemoveSignalGTK (GTK_OBJECT(entry), "activate");
@@ -2939,9 +2939,9 @@
 -------------------------------------------------------------------------*/
 void RemoveSignalGTK (GtkObject *w, gchar *signal_name)
 {
-  guint id;
+  intptr_t id;
   id = 0;
-  id = (guint)gtk_object_get_data (GTK_OBJECT (w), signal_name);
+  id = (intptr_t)gtk_object_get_data (GTK_OBJECT (w), signal_name);
   if (id)
     {
       gtk_signal_disconnect (GTK_OBJECT (w), id);
@@ -3789,7 +3789,7 @@
 	   gtk_object_set_data(GTK_OBJECT(statusbar), "MainSerie", 	   
 			       (gpointer)gtk_statusbar_get_context_id(GTK_STATUSBAR (statusbar), "MainSerie"));
 	   gtk_statusbar_push(GTK_STATUSBAR(statusbar),
-			      (guint)gtk_object_get_data (GTK_OBJECT(statusbar), "MainSerie"),
+			      (intptr_t)gtk_object_get_data (GTK_OBJECT(statusbar), "MainSerie"),
 			      "");
 	   
 	   /* The hbox which includes the logo and Comboboxzone (URL) */
@@ -5000,23 +5000,23 @@
 	(*(Proc3)ThotLocalActions[T_rcinsertpaste]) (
 		(void *)TRUE,
 		(void *)FALSE,
-		(void *)((int) data + 1) );
+		(void *)((intptr_t) data + 1) );
 	break;
       case NumMenuPaste:
 	(*(Proc3)ThotLocalActions[T_rcinsertpaste]) (
 		(void *)FALSE,
 		(void *)TRUE,
-	       	(void *)((int) data + 1));
+	       	(void *)((intptr_t) data + 1));
 	break;
       case NumMenuInclude:
 	(*(Proc3)ThotLocalActions[T_rcinsertpaste]) (
 		(void *)FALSE,
 		(void *)FALSE,
-	       	(void *)((int) data + 1));
+	       	(void *)((intptr_t) data + 1));
 	break;
       case NumMenuElChoice:
 	(*(Proc2)ThotLocalActions[T_rchoice]) (
-		(void *)((int) data + 1),
+		(void *)((intptr_t) data + 1),
 		(void *)NULL);
 	break;
       case NumSelectNatureName:
@@ -5031,7 +5031,7 @@
       case NumMenuAttrEnumNeeded:
 	(*(Proc3)ThotLocalActions[T_rattrreq]) (
 		(void *)ref,
-		(void *)((int) data),
+		(void *)((intptr_t) data),
 		(void *)data);
 	break;
       case NumMenuAttr:
@@ -5040,7 +5040,7 @@
       case NumMenuAttrEnum:
 	(*(Proc3)ThotLocalActions[T_rattrval]) (
 		(void *)ref,
-		(void *)((int) data),
+		(void *)((intptr_t) data),
 	       	(void *)data);
 	break;
 	
@@ -5054,7 +5054,7 @@
       case NumMenuAlphaLanguage:
 	(*(Proc3)ThotLocalActions[T_rattrlang]) (
 		(void *)ref,
-		(void *)((int) data),
+		(void *)((intptr_t) data),
 		(void *)NULL);
 	break;
       case NumFormClose:
@@ -5070,7 +5070,7 @@
       case NumMenuPaperFormat:
 	(*(Proc3)ThotLocalActions[T_rprint]) (
 		(void *)ref,
-		(void *)((int) data),
+		(void *)((intptr_t) data),
 		(void *)NULL);
 	break;
       case NumZonePrinterName:
@@ -5101,7 +5101,7 @@
       case NumTogglePatternUnchanged:
 	(*(Proc3)ThotLocalActions[T_present]) (
 		(void *)ref,
-		(void *)(int) data,
+		(void *)(intptr_t) data,
 		(void *)NULL);
 	break;
       case NumSelectPattern:
@@ -5116,7 +5116,7 @@
       case NumMenuPresentStandard:
 	(*(Proc2)ThotLocalActions[T_presentstd]) (
 		(void *)ref,
-		(void *)((int) data));
+		(void *)((intptr_t) data));
 	break;
       case NumFormSearchText:
       case NumMenuReplaceMode:
@@ -5125,7 +5125,7 @@
 	/* sous-menu mode de remplacement */
 	(*(Proc3)ThotLocalActions[T_searchtext]) (
 		(void *)ref,
-		(void *)((int) data),
+		(void *)((intptr_t) data),
 	       	(void *)NULL);
 	break;
       case NumZoneTextSearch:
@@ -5141,7 +5141,7 @@
       case NumMenuOrSearchText:
 	(*(Proc2)ThotLocalActions[T_locatesearch]) (
 		(void *)ref,
-		(void *)((int) data));
+		(void *)((intptr_t) data));
 	break;  
 	
       default:
@@ -5153,7 +5153,7 @@
 #endif /* #if defined(_GTK) || defined(_MOTIF) */
 	    (*(Proc3)ThotLocalActions[T_rattr]) (
 		(void *)ref,
-		(void *)((int) data),
+		(void *)((intptr_t) data),
 		(void *)ActiveFrame);
 	  }
 	break;
@@ -5187,7 +5187,7 @@
 #endif /* #if defined(_GTK) || defined(_MOTIF) */
 	      (*(Proc3)ThotLocalActions[T_rattr]) (
 			(void *)ref,
-			(void *)((int) data),
+			(void *)((intptr_t) data),
 			(void *)frame);
 	      return;
 	    }
@@ -5200,7 +5200,7 @@
 #endif /* #if defined(_GTK) || defined(_MOTIF) */
 	      (*(Proc3)ThotLocalActions[T_rselect]) (
 			(void *)ref,
-			(void *)((int) data + 1),
+			(void *)((intptr_t) data + 1),
 			(void *)frame);
 	      return;
 	    }
@@ -5230,7 +5230,7 @@
 
       if (ptrmenu)
 	{
-	  item = (int) data;
+	  item = (intptr_t) data;
 	  if (item < ptrmenu->ItemsNb && ptrmenu->ItemsList != NULL)
 	    {
 	      for (i = 0; i <= item; i++)
diff -Nur Amaya.bak/thotlib/dialogue/appli.c Amaya/thotlib/dialogue/appli.c
--- Amaya.bak/thotlib/dialogue/appli.c	2004-04-27 13:16:24.000000000 +0000
+++ Amaya/thotlib/dialogue/appli.c	2005-10-21 15:23:22.000000000 +0000
@@ -391,7 +391,7 @@
 #ifdef _GTK
 gboolean KillFrameGTK (GtkWidget *widget, GdkEvent *event, gpointer f)
 {
-  return KillFrameCallback( (int)f );
+  return KillFrameCallback( (intptr_t)f );
 }
 #endif /* _GTK */
 
@@ -852,13 +852,13 @@
   ----------------------------------------------------------------------*/
 gboolean FrameResizedGTK (GtkWidget *w, GdkEventConfigure *event, gpointer data)
 {
-  int            frame;
+  intptr_t            frame;
   int            width, height;
 #ifdef IV
   int         forever = 0;
 #endif /* IV */
  
-  frame = (int) data;
+  frame = (intptr_t) data;
   width = event->width;
   height = event->height; 
   if (width <= 0 || height <= 0 || frame == 0 || frame > MAX_FRAME)
@@ -933,11 +933,11 @@
   ----------------------------------------------------------------------*/
 gboolean ExposeCallbackGTK (ThotWidget widget, GdkEventExpose *event, gpointer data)
 {
-  int                  frame;
+  intptr_t            frame;
   unsigned int         x, y;
   unsigned int         w, h;
 
-  frame = (int) data;
+  frame = (intptr_t) data;
   x = event->area.x;
   y = event->area.y;
   w = event->area.width;
@@ -1955,9 +1955,9 @@
 	    else
 	      title_string = text;
 	    gtk_statusbar_pop (GTK_STATUSBAR(FrameTable[frame].WdStatus),
-			       (guint)gtk_object_get_data (GTK_OBJECT(FrameTable[frame].WdStatus), "MainSerie"));
+			       (intptr_t)gtk_object_get_data (GTK_OBJECT(FrameTable[frame].WdStatus), "MainSerie"));
 	    gtk_statusbar_push (GTK_STATUSBAR(FrameTable[frame].WdStatus),
-				(guint)gtk_object_get_data (GTK_OBJECT(FrameTable[frame].WdStatus), "MainSerie"),
+				(intptr_t)gtk_object_get_data (GTK_OBJECT(FrameTable[frame].WdStatus), "MainSerie"),
 				title_string);
 	    gtk_widget_show_all (GTK_WIDGET(FrameTable[frame].WdStatus));
 #endif /* _GTK */
@@ -2692,14 +2692,14 @@
 {
   Document            doc; 
   ViewFrame          *pFrame;
-  int                 frame;
+  intptr_t            frame;
   int                 view;
   static int          Motion_y = 0;
   static int          Motion_x = 0;
   GdkModifierType state = (GdkModifierType)GDK_BUTTON1_MOTION_MASK;
   int x,y;
   
-  frame = (int) data;
+  frame = (intptr_t) data;
   if (frame < 1)
     return FALSE;
 
@@ -3172,7 +3172,7 @@
   int                 comm, dx, dy, sel, h;
 #endif /* _MOTIF */
 #ifdef _GTK
-  int                 frame;
+  intptr_t            frame;
   GtkEntry            *textzone;
   static int          timer = None;
 #endif /* _GTK */
@@ -3180,7 +3180,7 @@
   View                view;
 
 #ifdef _GTK
-  frame = (int )data;
+  frame = (intptr_t )data;
 #endif /* _GTK */
 
 #ifdef _GL
@@ -3210,7 +3210,7 @@
 #endif /* _MOTIF */
   
 #ifdef _GTK
-    frame = (int )data;
+    frame = (intptr_t )data;
   if (FrameTable[frame].FrDoc == 0 ||
       documentDisplayMode[FrameTable[frame].FrDoc - 1] == NoComputedDisplay)
     /* don't manage a document with NoComputedDisplay mode */
diff -Nur Amaya.bak/thotlib/dialogue/browser.c Amaya/thotlib/dialogue/browser.c
--- Amaya.bak/thotlib/dialogue/browser.c	2004-01-29 16:06:45.000000000 +0000
+++ Amaya/thotlib/dialogue/browser.c	2005-10-21 15:25:20.000000000 +0000
@@ -38,7 +38,7 @@
   ----------------------------------------------------------------------*/
 void TtaExtractSuffix (char *aName, char *aSuffix)
 {
-  int               lg, i;
+  intptr_t          lg, i;
   char             *ptr, *oldptr;
 
   if (!aSuffix || !aName)
@@ -59,7 +59,7 @@
 	}
       while (ptr);
 
-      i = (int) (oldptr) - (int) (aName);	/* name length */
+      i = (intptr_t) (oldptr) - (intptr_t) (aName);	/* name length */
       if (i > 1)
 	{
 	  aName[i - 1] = EOS;
diff -Nur Amaya.bak/thotlib/dialogue/callback.c Amaya/thotlib/dialogue/callback.c
--- Amaya.bak/thotlib/dialogue/callback.c	2004-04-14 14:29:32.000000000 +0000
+++ Amaya/thotlib/dialogue/callback.c	2005-10-21 15:17:40.000000000 +0000
@@ -409,7 +409,7 @@
   /* A menu entry is selected */
   if (catalogue->Cat_Widget)
     {
-      if ((int) catalogue->Cat_Widget == -1)
+      if ((intptr_t) catalogue->Cat_Widget == -1)
 	/*** back to a simple button ***/
 	(*(Proc3)CallbackDialogueProc) (
 		(void *)catalogue->Cat_Ref,
@@ -597,15 +597,15 @@
 ThotBool CallRadioGTK (ThotWidget w, struct Cat_Context *catalogue)
 {
   register int        i;
-  register int        index;
-  register int        entry;
+  register intptr_t   index;
+  register intptr_t   entry;
   struct E_List      *adbloc;
 
   if (gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (w)) == FALSE)
     {
       /* Prevent to unselect an element directly...
 	 you must select another one to unselect others  */
-      index = (guint) gtk_object_get_data (GTK_OBJECT (w), "toggled");
+      index = (intptr_t) gtk_object_get_data (GTK_OBJECT (w), "toggled");
       gtk_signal_handler_block (GTK_OBJECT(w), index);
       gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (w), TRUE);
       gtk_signal_handler_unblock (GTK_OBJECT(w), index); 
@@ -641,7 +641,7 @@
 	    {
 	      if (adbloc->E_ThotWidget[i] != w) 
 		{
-		  index = (guint) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]), "toggled");
+		  index = (intptr_t) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]), "toggled");
 		  gtk_signal_handler_block (GTK_OBJECT(adbloc->E_ThotWidget[i]), index);
 		  
 		  gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON (adbloc->E_ThotWidget[i]), FALSE);
@@ -1073,7 +1073,7 @@
   ----------------------------------------------------------------------*/
 void CallValueSet (ThotWidget w, struct Cat_Context *catalogue, caddr_t call_d)
 {
-  int               val, val1;
+  intptr_t          val, val1;
   char              text[11];
   ThotWidget        wtext;
 
@@ -1097,10 +1097,10 @@
 	  else
 	    sscanf (text, "%d", &val);
 	  /* Est-ce une valeur valide ? */
-	  if (val < (int) catalogue->Cat_Entries->E_ThotWidget[2])
-	    val1 = (int) catalogue->Cat_Entries->E_ThotWidget[2];
-	  else if (val > (int) catalogue->Cat_Entries->E_ThotWidget[3])
-	    val1 = (int) catalogue->Cat_Entries->E_ThotWidget[3];
+	  if (val < (intptr_t) catalogue->Cat_Entries->E_ThotWidget[2])
+	    val1 = (intptr_t) catalogue->Cat_Entries->E_ThotWidget[2];
+	  else if (val > (intptr_t) catalogue->Cat_Entries->E_ThotWidget[3])
+	    val1 = (intptr_t) catalogue->Cat_Entries->E_ThotWidget[3];
 	  else
 	    val1 = val;	/* valeur inchangee */
 	  
diff -Nur Amaya.bak/thotlib/dialogue/closedoc.c Amaya/thotlib/dialogue/closedoc.c
--- Amaya.bak/thotlib/dialogue/closedoc.c	2004-01-29 16:06:45.000000000 +0000
+++ Amaya/thotlib/dialogue/closedoc.c	2005-10-21 15:25:57.000000000 +0000
@@ -50,7 +50,7 @@
   ----------------------------------------------------------------------*/
 void CallbackCloseDocMenu (int ref, int typedata, char *data)
 {
-  switch ((int) data)
+  switch ((intptr_t) data)
     {
     case 0:
       /* abandon du formulaire */
diff -Nur Amaya.bak/thotlib/dialogue/colors.c Amaya/thotlib/dialogue/colors.c
--- Amaya.bak/thotlib/dialogue/colors.c	2004-01-29 16:06:45.000000000 +0000
+++ Amaya/thotlib/dialogue/colors.c	2005-10-21 14:59:51.000000000 +0000
@@ -212,7 +212,7 @@
   ----------------------------------------------------------------------*/
 gboolean ColorsExposeGTK (GtkWidget *widget, GdkEventExpose *ev, gpointer data)
 {
-  int color = (int) data;
+  intptr_t color = (intptr_t) data;
   GdkWindow *window = GTK_WIDGET(widget)->window;
   gdk_rgb_gc_set_foreground (TtLineGC, ColorPixel (color));
   gdk_draw_rectangle (window, TtLineGC, TRUE, 0, 0, 20, 20);
@@ -279,7 +279,7 @@
   ----------------------------------------------------------------------*/
 gboolean ColorsPressGTK (GtkWidget *widget, gpointer data)
 {
-  int color = (int)data;
+  intptr_t color = (intptr_t)data;
   
   if (ApplyFg)
     {
diff -Nur Amaya.bak/thotlib/dialogue/dialogapi.c Amaya/thotlib/dialogue/dialogapi.c
--- Amaya.bak/thotlib/dialogue/dialogapi.c	2004-04-27 13:16:25.000000000 +0000
+++ Amaya/thotlib/dialogue/dialogapi.c	2005-10-21 15:28:46.000000000 +0000
@@ -3770,7 +3770,7 @@
    Arg                 args[MAX_ARGS];
 #endif /* #ifdef _MOTIF */
 #ifdef _GTK
-   guint               id_toggled;
+   intptr_t             id_toggled;
 #endif /* _GTK */
    struct Cat_Context *catalogue;
 
@@ -3827,7 +3827,7 @@
 #endif /* _MOTIF */
 #ifdef _GTK
 
-		       id_toggled  = (guint) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]),
+		       id_toggled  = (intptr_t) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]),
 							  "toggled");
 		       gtk_signal_handler_block (GTK_OBJECT(adbloc->E_ThotWidget[i]), 
 						 id_toggled);
@@ -3845,7 +3845,7 @@
 		       XtSetValues (adbloc->E_ThotWidget[i], args, n);
 #endif /* _MOTIF */
 #ifdef _GTK
-		       id_toggled  = (guint) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]),
+		       id_toggled  = (intptr_t) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]),
 							  "toggled");
 		       gtk_signal_handler_block (GTK_OBJECT(adbloc->E_ThotWidget[i]), 
 						 id_toggled);
@@ -3876,7 +3876,7 @@
 		  i = C_NUMBER;
 		else
 		  {
-		    id_toggled  = (guint) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]),
+		    id_toggled  = (intptr_t) gtk_object_get_data (GTK_OBJECT (adbloc->E_ThotWidget[i]),
 							       "toggled");
 		    gtk_signal_handler_unblock (GTK_OBJECT(adbloc->E_ThotWidget[i]), 
 						id_toggled); 
@@ -6023,8 +6023,8 @@
   
 #ifdef _GTK
 	/* delete the old list elements */
-	if ((gint)gtk_object_get_data(GTK_OBJECT(w), "GList") > 0)
-	  gtk_list_clear_items (GTK_LIST (w), 0, (gint)gtk_object_get_data(GTK_OBJECT (w), "GList"));
+	if ((intptr_t)gtk_object_get_data(GTK_OBJECT(w), "GList") > 0)
+	  gtk_list_clear_items (GTK_LIST (w), 0, (intptr_t)gtk_object_get_data(GTK_OBJECT (w), "GList"));
 	/* add the new list elements */
 	gtk_list_append_items (GTK_LIST(w), item);
 	/* update the number of list element */
@@ -7332,8 +7332,8 @@
 	     return;
 	  }
 	/* Est-ce une valeur valide ? */
-	if (val < (int) catalogue->Cat_Entries->E_ThotWidget[2]
-	    || val > (int) catalogue->Cat_Entries->E_ThotWidget[3])
+	if (val < (intptr_t) catalogue->Cat_Entries->E_ThotWidget[2]
+	    || val > (intptr_t) catalogue->Cat_Entries->E_ThotWidget[3])
 	  {
 	     TtaError (ERR_invalid_reference);
 	     return;
diff -Nur Amaya.bak/thotlib/dialogue/font.c Amaya/thotlib/dialogue/font.c
--- Amaya.bak/thotlib/dialogue/font.c	2004-04-27 13:16:25.000000000 +0000
+++ Amaya/thotlib/dialogue/font.c	2005-10-21 14:32:13.000000000 +0000
@@ -948,7 +948,7 @@
   ----------------------------------------------------------------------*/
 int PixelValue (int val, TypeUnit unit, PtrAbstractBox pAb, int zoom)
 {
-  int              dist, i;
+  intptr_t              dist, i;
 
   dist = 0;
   switch (unit)
@@ -1003,7 +1003,7 @@
 #endif /* _WIN_PRINT */
       break;
     case UnPercent:
-      i = val * (int) pAb;
+      i = val * (intptr_t) pAb;
       dist = (i + 50) / 100;
       break;
     case UnAuto:
@@ -1020,7 +1020,7 @@
   ----------------------------------------------------------------------*/
 int LogicalValue (int val, TypeUnit unit, PtrAbstractBox pAb, int zoom)
 {
-   int              dist, i;
+   intptr_t              dist, i;
 
    dist = 0;
    switch (unit)
@@ -1080,7 +1080,7 @@
        else
 	 {
 	   i = val * 100;
-	   dist = i / (int) pAb;
+	   dist = i / (intptr_t) pAb;
 	 }
        break;
      case UnAuto:
diff -Nur Amaya.bak/thotlib/dialogue/input.c Amaya/thotlib/dialogue/input.c
--- Amaya.bak/thotlib/dialogue/input.c	2004-04-27 13:16:25.000000000 +0000
+++ Amaya/thotlib/dialogue/input.c	2005-10-21 15:30:22.000000000 +0000
@@ -650,9 +650,9 @@
   unsigned char       value;
   int                 status;
   int                 PicMask;
-  int                 frame;
+  intptr_t            frame;
 
-  frame = (int) data;
+  frame = (intptr_t) data;
   if (frame > MAX_FRAME)
     frame = 0;
   FrameToView (frame, &document, &view);
@@ -737,11 +737,11 @@
 ----------------------------------------------------------------------*/
 gboolean GtkLiningDown (gpointer data)
 {
-  int       frame;
+  intptr_t  frame;
   Document  doc; 
   int       view;
   
-   frame = (int) data; 
+   frame = (intptr_t) data; 
    FrameToView (frame, &doc, &view);
    TtcLineDown (doc, view);
    /* As this is a timeout function, return TRUE so that it
@@ -753,11 +753,11 @@
 ----------------------------------------------------------------------*/
 gboolean GtkLiningUp (gpointer data)
 {
-  int       frame;
+  intptr_t  frame;
   Document  doc; 
   int       view;
   
-  frame = (int) data; 
+  frame = (intptr_t) data; 
   FrameToView (frame, &doc, &view);
   TtcLineUp (doc, view);
   /* As this is a timeout function, return TRUE so that it
@@ -785,7 +785,7 @@
   GdkModifierType state;
   GtkEntry           *textzone;
 
-  frame = (int) data; 
+  frame = (intptr_t) data; 
   FrameToView (frame, &doc, &view);
   textzone = 0;
   if (timer != None)
diff -Nur Amaya.bak/thotlib/dialogue/printmenu.c Amaya/thotlib/dialogue/printmenu.c
--- Amaya.bak/thotlib/dialogue/printmenu.c	2004-01-29 16:06:45.000000000 +0000
+++ Amaya/thotlib/dialogue/printmenu.c	2005-10-21 15:32:04.000000000 +0000
@@ -343,7 +343,7 @@
        printArgc++;
 #else /* _WINDOWS_DLL */
 #ifndef _WX
-       sprintf (&cmd[i], " -w%u", (unsigned int) FrRef[0]);
+       sprintf (&cmd[i], " -w%lu", (intptr_t) FrRef[0]);
 #else /* _WX */
        sprintf (&cmd[i], " -w%u", (unsigned int) FrameTable[0].WdFrame);
 #endif /* _WX */
@@ -362,9 +362,9 @@
 	   printArgc++;
 #else /* _WINDOWS_DLL */
 #ifndef _WX
-           sprintf (&cmd[i], " -w%u", (unsigned int) FrRef[0]);
+           sprintf (&cmd[i], " -w%u", (intptr_t) FrRef[0]);
 #else /* _WX */
-           sprintf (&cmd[i], " -w%u", (unsigned int) FrameTable[0].WdFrame);
+           sprintf (&cmd[i], " -w%u", (intptr_t) FrameTable[0].WdFrame);
 #endif /* _WX */
 #endif /* _WINDOWS_DLL */
 	 }
@@ -376,9 +376,9 @@
 	   printArgc++;
 #else /* _WINDOWS_DLL */
 #ifndef _WX
-           sprintf (&cmd[i], " -w%u", (unsigned int) FrRef[0]);
+           sprintf (&cmd[i], " -w%u", (intptr_t) FrRef[0]);
 #else /* _WX */
-           sprintf (&cmd[i], " -w%u", (unsigned int) FrameTable[0].WdFrame);
+           sprintf (&cmd[i], " -w%u", (intptr_t) FrameTable[0].WdFrame);
 #endif /* _WX */
 #endif /* _WINDOWS_DLL */
 	 }
diff -Nur Amaya.bak/thotlib/editing/checkermenu.c Amaya/thotlib/editing/checkermenu.c
--- Amaya.bak/thotlib/editing/checkermenu.c	2004-02-25 10:06:16.000000000 +0000
+++ Amaya/thotlib/editing/checkermenu.c	2005-10-21 15:33:34.000000000 +0000
@@ -718,7 +718,7 @@
     switch (ref - SpellingBase)
       {
       case ChkrMenuIgnore:
-	switch ((int) data)
+	switch ((intptr_t) data)
 	  {
 	  case 0:
 	    /* ignorer les mots ecrits en majuscule */
@@ -752,7 +752,7 @@
 	break;
       case ChkrCaptureNC:
 	/* modification de Nb de corrections a proposer */
-	NC = (int) data;
+	NC = (intptr_t) data;
 	if (NC > OldNC && ChkrElement != NULL && CurrentWord[0] != EOS)
 	  SetProposals (ChkrLanguage);
 	else
@@ -760,7 +760,7 @@
 	break;
       case ChkrMenuOR:
 	/* definition du sens de correction OU? */
-	switch ((int) data)
+	switch ((intptr_t) data)
 	  {
 	  case 0:	/* ElemIsBefore la selection */
 	    ChkrRange->SStartToEnd = FALSE;
@@ -778,7 +778,7 @@
 	/* La premiere fois on initialise le domaine de recherche */
 	if (FirstStep)
 	  {
-	    InitSearchDomain ((int) data, ChkrRange);
+	    InitSearchDomain ((intptr_t) data, ChkrRange);
 	    /* On prepare la recheche suivante */
 #ifndef _WINGUI
 	    if (ChkrRange->SStartToEnd)
@@ -816,7 +816,7 @@
       case ChkrFormCorrect:
 	/* retour de la feuille de dialogue CORRECTION */
 	/* effectuer l'action demandee */
-	ApplyCommand ((int) data);
+	ApplyCommand ((intptr_t) data);
 	break;
       }
 }
Binary files Amaya.bak/thotlib/image/.xpmcreate.c.swp and Amaya/thotlib/image/.xpmcreate.c.swp differ
diff -Nur Amaya.bak/thotlib/internals/h/XpmI.h Amaya/thotlib/internals/h/XpmI.h
--- Amaya.bak/thotlib/internals/h/XpmI.h	1999-09-29 16:36:33.000000000 +0000
+++ Amaya/thotlib/internals/h/XpmI.h	2005-10-21 15:09:52.000000000 +0000
@@ -225,7 +225,7 @@
 FUNC(xpmHashIntern, int, (xpmHashTable *table, char *tag, void *data));
 
 #define HashAtomData(i) ((void *)i)
-#define HashColorIndex(slot) ((unsigned int)((*slot)->data))
+#define HashColorIndex(slot) ((intptr_t)((*slot)->data))
 #define USE_HASHTABLE (cpp > 2 && ncolors > 4)
 
 /* I/O utility */
diff -Nur Amaya.bak/thotlib/presentation/style.c Amaya/thotlib/presentation/style.c
--- Amaya.bak/thotlib/presentation/style.c	2004-04-14 14:29:33.000000000 +0000
+++ Amaya/thotlib/presentation/style.c	2005-10-21 15:13:39.000000000 +0000
@@ -810,7 +810,8 @@
   char               *attrVal;
   unsigned int        elementType;
   int                 nbrules;
-  int                 i, j, val;
+  int                 i, j;
+  intptr_t            val;
   unsigned int        match;
   
 
@@ -874,7 +875,7 @@
 	      ppRule = &(attrs->ApRefFirstPRule);
 	      break;
 	    case AtEnumAttr:
-	      val = (int) attrVal;
+	      val = (intptr_t) attrVal;
 	      if (val)
 		ppRule = &(attrs->ApEnumFirstPRule[val]);
 	      else
@@ -900,7 +901,8 @@
   PtrSSchema          pSS;
   PtrPRule           *ppRule;
   char               *attrVal;
-  int                 nbrules, val, match;
+  int                 nbrules, match;
+  intptr_t            val;
 
   pSS = (PtrSSchema) ctxt->schema;
   ppRule = FirstPresAttrRuleSearch (tsch, attrType, ctxt, att, &attrs);
@@ -969,7 +971,7 @@
 	  break;
 	case AtEnumAttr:
 	  /* get the attribute value */
-	  val = (int) attrVal;
+	  val = (intptr_t) attrVal;
 	  if (val > 0)
 	    {
 	      new_->ApEnumFirstPRule[val] = NULL;
