--- amaya-9.4.orig/Amaya/amaya/Makefile.in
+++ amaya-9.4Amaya/amaya/Makefile.in
@@ -508,10 +508,10 @@
 @NOPRINT@	@(cd ../thotlib ; $(MAKE) CC="$(CC)" CFLAGS="$(CFLAGS)" CXX="$(CXX)" CXXFLAGS="$(CXXFLAGS)" print)
 
 install : all @NOPRINT@../bin/print
-	../script_install ../bin $(prefix) $(bindir)
+	../script_install ../bin $(DESTDIR)$(prefix) $(DESTDIR)$(bindir)
 
 uninstall :
-	../script_uninstall $(prefix) $(bindir)
+	../script_uninstall $(DESTDIR)$(prefix) $(DESTDIR)$(bindir)
 
 CEXTRACT= ../bin/cextract
 
--- amaya-9.4.orig/Amaya/thotlib/include/Makefile.in
+++ amaya-9.4/Amaya/thotlib/include/Makefile.in
@@ -24,14 +24,14 @@
 clean :
 
 install :
-	@(if test ! -d $(includedir) ; then $(MKDIR) $(includedir); fi)
-	@(if test ! -d $(includedir)/thot ; then $(MKDIR) $(includedir)/thot; fi)
+	@(if test ! -d $(DESTDIR)$(includedir) ; then $(MKDIR) $(DESTDIR)$(includedir); fi)
+	@(if test ! -d $(DESTDIR)$(includedir)/thot ; then $(MKDIR) $(DESTDIR)$(includedir)/thot; fi)
 	(for include in $(INCLUDES) ; do \
-	  $(INSTALL_DATA) $(THOTDIR)/thotlib/include/$$include $(includedir)/thot ; \
+	  $(INSTALL_DATA) $(THOTDIR)/thotlib/include/$$include $(DESTDIR)$(includedir)/thot ; \
 	  done)
 
 uninstall :
-	$(RM) -rf $(includedir)/thot
+	$(RM) -rf $(DESTDIR)$(includedir)/thot
 
 depend :
 
