Index: amaya-9.4/Amaya/Makefile.in
===================================================================
--- amaya-9.4.orig/Amaya/Makefile.in	2006-03-12 20:10:37.959334400 +0000
+++ amaya-9.4/Amaya/Makefile.in	2006-03-12 20:13:41.420444072 +0000
@@ -201,6 +201,7 @@
 	  cd $(LIBWWW) ; unset LANG; \
 	  if [ ! -f Makefile -o ! -f wwwconf.h ] ; then \
 	  	$$libwwwdir/configure \
+			--build=@build_alias@ --host=@host_alias@ --target=@target_alias@ \
 			--disable-shared \
 			@WITHDAV@ \
 			--with-zlib; \
@@ -313,6 +314,7 @@
 	  if [ ! -f Makefile -o ! -f config.h ] ; then \
 	  	CFLAGS="-I$$raptordir/expat/lib" \
 			$$raptordir/configure \
+			--build=@build_alias@ --host=@host_alias@ --target=@target_alias@ \
 			--without-shared \
 			--with-expat ; \
 		$(ECHO) "patching libraptor/libtool because -flat_namespace option is missing and do not compile on macosx" ; \
@@ -365,6 +367,7 @@
 			CPPFLAGS="-I$$localdir/../../$(LIBWWW)/modules/expat/lib" \
 			LDFLAGS="-L$$localdir/libwww/modules/expat -L$$localdir/libwww/modules/expat/.libs" \
 			$$redlanddir/configure \
+			--build=@build_alias@ --host=@host_alias@ --target=@target_alias@ \
 			--disable-shared \
 			--without-bdb \
 	 		--enable-parsers=raptor \
Index: amaya-9.4/Amaya/configure.in
===================================================================
--- amaya-9.4.orig/Amaya/configure.in	2006-03-12 20:10:37.961334096 +0000
+++ amaya-9.4/Amaya/configure.in	2006-03-12 20:12:20.023818240 +0000
@@ -472,7 +472,7 @@
     #   --enable-unicode        compile wxString with Unicode support
     #   --with-gtk              use GTK+
     #   --with-opengl           use OpenGL (or Mesa)
-    WXCONFIGURE_OPTION="--with-opengl --with-gtk --enable-gtk2 --enable-unicode --enable-ipc --disable-shared  --with-libpng=builtin --with-libjpeg=builtin --with-libtiff=builtin $WXCONFIGURE_TUNNING_OPTION"
+    WXCONFIGURE_OPTION="--build=$build_alias --host=$host_alias --target=$target_alias --with-opengl --with-gtk --enable-gtk2 --enable-unicode --enable-ipc --disable-shared  --with-libpng=builtin --with-libjpeg=builtin --with-libtiff=builtin $WXCONFIGURE_TUNNING_OPTION"
   fi
   if test "$AMAYAOS" = "MACOSX" ; then
     # MACOSX
