Applied upstream

Index: amaya-9.51/Amaya/thotlib/dialogue/AmayaXMLPanel.cpp
===================================================================
--- amaya-9.51.orig/Amaya/thotlib/dialogue/AmayaXMLPanel.cpp	2006-05-17 19:09:52.587516720 +0100
+++ amaya-9.51/Amaya/thotlib/dialogue/AmayaXMLPanel.cpp	2006-05-17 19:11:42.397823016 +0100
@@ -102,7 +102,7 @@
   int nb_el                 = (int)p.param1;
   const char * listBuffer   = (char *)p.param2;
   const char * currentEl    = (char *)p.param3;
-  int ref                   = (int)p.param4;;
+  intptr_t ref              = (intptr_t)p.param4;;
   
   m_XMLRef = ref;
   
