Index: amaya-9.53/amaya/MENUconf.c
===================================================================
--- amaya-9.53.orig/amaya/MENUconf.c	2007-01-10 23:25:15.000000000 +0000
+++ amaya-9.53/amaya/MENUconf.c	2007-01-10 23:37:07.000000000 +0000
@@ -4369,15 +4369,15 @@
   ----------------------------------------------------------------------*/
 static void TemplatesCallbackDialog (int ref, int typedata, char *data)
 {
-    printf("TemplatesCallbackDialog : %d %d (%d)\n", ref, ref-TemplatesBase, (int)data);
+    printf("TemplatesCallbackDialog : %d %d (%d)\n", ref, ref-TemplatesBase, (uintptr_t)data);
 
-    long val;
+    uintptr_t val;
     if (ref==-1)
     {
     }
     else
     {
-        val = (long int) data;
+        val = (uintptr_t) data;
         switch (ref - TemplatesBase)
         {
             case TemplatesMenu:
