Index: Amaya/davlib/Makefile.in
===================================================================
--- Amaya.orig/davlib/Makefile.in	2008-09-19 22:10:52.000000000 +0100
+++ Amaya/davlib/Makefile.in	2008-09-19 22:22:13.000000000 +0100
@@ -13,38 +13,14 @@
 
 INCLUDES =-DHAVE_CONFIG_H -I.. \
 	  -I../amaya -I@srcdir@/f -I@top_srcdir@/amaya \
-	  -I@top_srcdir@/amaya/f -I../libwww \
+	  -I@top_srcdir@/amaya/f \
+	  @LIBWWW_CFLAGS@ \
 	  -I@top_srcdir@/thotlib/include \
-	  -I@top_srcdir@/../libwww/Library/src \
-	  -I@top_srcdir@/../libwww/modules/expat/lib \
 	  -I$(DAVLIB)/tree/h -I$(DAVLIB)/h \
 	  $(GUI_INCLUDES)
 
 OPTIONS	= @AMAYA_OPTIONS@ $(AMAYA_DAV_OPTIONS)
 
-LIBWWW_LIBS= \
-		-L../$(LIBWWW)/Library/src/.libs \
-			-lwwwapp \
-			-lwwwhttp \
-			-lwwwxml \
-			-lwwwhtml \
-			-lwwwmime \
-			-lwwwcache \
-			-lwwwstream \
-			-lwwwfile \
-			-lwwwdir \
-			-lwwwtrans \
-			-lwwwcore \
-			-lwwwutils \
-			-lwwwzip \
-			-lwwwftp \
-			-lwwwdav \
-		-L../$(LIBWWW)/modules/md5/.libs \
-			-lmd5 \
-		-L../$(LIBWWW)/modules/expat/.libs \
-			-lexpat \
-		-ldl -L/usr/lib
-
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -52,7 +28,7 @@
 
 ############################################
 
-all : tree awparser.o AHTLockBase.o HTHome dav_obj davlib.a
+all : tree awparser.o AHTLockBase.o dav_obj davlib.a
 
 dav_obj : $(AMAYA_DAV_OBJ)
 
@@ -108,8 +84,7 @@
 	awnode.o \
 	awstring.o \
 	awpair.o \
-	list.o \
-	../$(LIBWWW)/Library/src/HTHome.o
+	list.o
 
 #
 # dependencies
Index: Amaya/annotlib/Makefile.in
===================================================================
--- Amaya.orig/annotlib/Makefile.in	2008-09-19 22:10:52.000000000 +0100
+++ Amaya/annotlib/Makefile.in	2008-09-19 22:22:13.000000000 +0100
@@ -14,9 +14,8 @@
 
 INCLUDES= -DHAVE_CONFIG_H -DANNOTATIONS \
 	  -I.. -I../amaya  -I@srcdir@/f -I@top_srcdir@/amaya \
-	  -I@top_srcdir@/amaya/f -I../libwww \
-	  -I@top_srcdir@/../libwww/Library/src \
-	  -I@top_srcdir@/../libwww/modules/expat/lib \
+	  -I@top_srcdir@/amaya/f \
+	   @LIBWWW_CFLAGS@ \
 	   @AMAYA_RAPTOR_INCLUDES@ @THOTINCLUDES@ $(GUI_INCLUDES)
 
 
Index: Amaya/configure.in
===================================================================
--- Amaya.orig/configure.in	2008-09-19 22:12:07.000000000 +0100
+++ Amaya/configure.in	2008-09-19 22:22:13.000000000 +0100
@@ -689,6 +689,51 @@
 	esac],
 	[with_svg="yes"])
 
+dnl
+dnl WebDAV support enabled by default
+dnl
+AC_ARG_WITH(dav,
+  AC_HELP_STRING([--with-dav],[enable WebDAV support (default)]),
+  [case "${withval}" in
+	  yes) with_dav="yes" ;;
+	  no)  with_dav="no" ;;
+	  *) AC_MSG_ERROR(bad value ${withval} for --with-dav) ;;
+	esac],
+	[with_dav="yes"])
+
+# If DAV is enabled, check if the source code is present
+if test ! -f $srcdir/davlib/Makefile.in ; then
+	if test "$with_dav" = "yes" ; then
+	    AC_MSG_WARN(WebDAV makefile not found, disabling WebDAV build)
+	    with_dav="no"
+	fi
+fi
+
+if test "$with_dav" = "yes" ; then
+   DAV_OPTIONS=-DDAV
+   DAVDIR="davlib"
+   LIBDAV="libwwwdav"
+   WITHDAV="--with-dav" 
+   WWWDAV="-lwwwdav" 
+else
+   with_dav="no"
+   DAV_OPTIONS=""
+   DAVDIR=""
+   LIBDAV=""
+   WITHDAV="" 
+   WWWDAV="" 
+fi
+
+CURRENTPATH="`pwd`"
+LIBWWW_CFLAGS="-I$CURRENTPATH/../../libwww/Library/src -I$CURRENTPATH/../../libwww/modules/expat/lib -I$CURRENTPATH/../../libwww"
+
+AC_SUBST(LIBWWW_CFLAGS)
+
+AC_SUBST(DAVDIR)
+AC_SUBST(LIBDAV)
+AC_SUBST(WITHDAV)
+AC_SUBST(WWWDAV)
+
 ##########################Annotations Support####################
 dnl
 dnl annotations are enabled by default
@@ -833,28 +878,6 @@
  GENERIC_XML_OPTIONS=""
 fi
 
-dnl
-dnl WebDAV support enabled by default
-dnl
-AC_ARG_WITH(dav,            [  --with-dav              enable WebDAV support (default) ])
-if test "$with_dav" = "" ; then
-  with_dav="yes"
-fi
-if test "$with_dav" = "yes" ; then
-   DAV_OPTIONS=-DDAV
-   DAVDIR="davlib"
-   LIBDAV="libwwwdav"
-   WITHDAV="--with-dav" 
-   WWWDAV="-lwwwdav" 
-else
-   with_dav="no"
-   DAV_OPTIONS=""
-   DAVDIR=""
-   LIBDAV=""
-   WITHDAV="" 
-   WWWDAV="" 
-fi
-
 
 if test "$build_amaya" = "yes" ; then
     if test ! -f $srcdir/../libwww/Makefile.in ; then
@@ -881,19 +904,6 @@
 	fi
     fi
 
-    if test ! -f $srcdir/davlib/Makefile.in ; then
-	if test "$with_dav" = "yes" ; then
-	    AC_MSG_WARN(WebDAV makefile not found !!)
-	    AC_MSG_WARN(Disabling WebDAV build !)
-	    with_dav="no"
-            DAVDIR=""
-            LIBDAV=""
-            WITHDAV="" 
-            WWWDAV="" 
-	fi
-    fi
-   
-
 fi
 
 dnl
@@ -988,11 +998,6 @@
 AC_SUBST(AMAYA_OPTIONS)
 AC_SUBST(EXTRA_INCLUDES)
 
-AC_SUBST(DAVDIR)
-AC_SUBST(LIBDAV)
-AC_SUBST(WITHDAV)
-AC_SUBST(WWWDAV)
-
 AC_SUBST(MAKE_LIBWWW_RDF_PARSER)
 AC_SUBST(LIBWWW_RDF_PARSER)
 #AC_SUBST(ANNOTLIB_COMPILE_BM)
Index: Amaya/amaya/Makefile.math
===================================================================
--- Amaya.orig/amaya/Makefile.math	2008-09-19 22:10:52.000000000 +0100
+++ Amaya/amaya/Makefile.math	2008-09-19 22:22:13.000000000 +0100
@@ -23,7 +23,7 @@
 
 MATH_SCHEMAS= $(AMAYA)/MathML.S $(AMAYA)/MathMLP.P $(AMAYA)/MathMLT.T
 
-math_schema= compMATH compMATHP compMATHT
+math_schema= compMATH compMATHP compMATHT MathML.h
 
 #
 # Structure schema compilation.
Index: Amaya/amaya/Makefile.svg
===================================================================
--- Amaya.orig/amaya/Makefile.svg	2008-09-19 22:10:52.000000000 +0100
+++ Amaya/amaya/Makefile.svg	2008-09-19 22:22:13.000000000 +0100
@@ -31,7 +31,7 @@
 	$(AMAYA)/Timeline.S $(AMAYA)/TimelineP.P $(AMAYA)/TimelineT.T
 
 svg_schema= compSVG compSVGP compSVGT \
-	compTimeline compTimelineP compTimelineT
+	compTimeline compTimelineP compTimelineT SVG.h
 
 
 #
