--- libwww/configure.ac.orig	2003-06-01 08:54:16.000000000 -0700
+++ libwww/configure.ac	2003-06-01 08:54:28.000000000 -0700
@@ -37,7 +37,7 @@
 ])
 
 dnl taken from the DDD configure.in
-AC_DEFUN(AC_STRUCT_WINSIZE,
+AC_DEFUN([AC_STRUCT_WINSIZE],
 [AC_MSG_CHECKING(for struct winsize in sys/ioctl.h and termios.h)
 AC_CACHE_VAL(libwww_cv_struct_winsize_header,
 [AC_TRY_COMPILE([#include <sys/types.h>
