Index: amaya-9.55~dfsg.0/configure.in
===================================================================
--- amaya-9.55~dfsg.0.orig/configure.in	2007-03-31 13:51:41.000000000 +0100
+++ amaya-9.55~dfsg.0/configure.in	2007-03-31 13:51:48.000000000 +0100
@@ -1433,7 +1433,7 @@
 make_output="Makefile Options.orig:Options.in tools/Makefile \
        libjpeg/Makefile libpng/Makefile libpng/zlib/Makefile \
        tools/mkdep/Makefile \
-       tools/cextract-1.7/Makefile thotlib/Makefile \
+       thotlib/Makefile \
        thotlib/include/Makefile batch/Makefile \
        script_install:../batch/script_install.in \
        script_install_gnomekde:../batch/script_install_gnomekde.in \
Index: amaya-9.55~dfsg.0/Makefile.in
===================================================================
--- amaya-9.55~dfsg.0.orig/Makefile.in	2007-03-31 13:51:41.000000000 +0100
+++ amaya-9.55~dfsg.0/Makefile.in	2007-03-31 13:51:48.000000000 +0100
@@ -195,8 +195,7 @@
 $(THOTDIR)/thotlib/Makefile.in \
 $(THOTDIR)/thotlib/include/Makefile.in \
 $(THOTDIR)/tools/Makefile.in \
-$(THOTDIR)/tools/mkdep/Makefile.in \
-$(THOTDIR)/tools/cextract-1.7/Makefile.in
+$(THOTDIR)/tools/mkdep/Makefile.in
 
 rebuild : tools/mkdep/mkdep Makefile
 
