Index: amaya-9.53~dfsg.0/configure.in
===================================================================
--- amaya-9.53~dfsg.0.orig/configure.in	2007-01-12 23:35:10.000000000 +0000
+++ amaya-9.53~dfsg.0/configure.in	2007-01-12 23:35:33.000000000 +0000
@@ -1575,7 +1575,7 @@
 make_output="Makefile Options.orig:Options.in tools/Makefile \
        libjpeg/Makefile libpng/Makefile libpng/zlib/Makefile \
        tools/mkdep/Makefile \
-       tools/cextract-1.7/Makefile thotlib/Makefile \
+       thotlib/Makefile \
        thotlib/include/Makefile batch/Makefile \
        script_install:../batch/script_install.in \
        script_install_gnomekde:../batch/script_install_gnomekde.in \
Index: amaya-9.53~dfsg.0/tools/Makefile.in
===================================================================
--- amaya-9.53~dfsg.0.orig/tools/Makefile.in	2007-01-12 23:35:43.000000000 +0000
+++ amaya-9.53~dfsg.0/tools/Makefile.in	2007-01-12 23:35:52.000000000 +0000
@@ -7,7 +7,7 @@
 
 include ../Options
 
-SUBDIRS = mkdep dns_daemon javastub cextract-1.7
+SUBDIRS = mkdep dns_daemon javastub
 
 @SET_MAKE@
 
Index: amaya-9.53~dfsg.0/Makefile.in
===================================================================
--- amaya-9.53~dfsg.0.orig/Makefile.in	2007-01-13 00:05:39.000000000 +0000
+++ amaya-9.53~dfsg.0/Makefile.in	2007-01-13 00:05:55.000000000 +0000
@@ -235,8 +235,7 @@
 $(THOTDIR)/thotlib/Makefile.in \
 $(THOTDIR)/thotlib/include/Makefile.in \
 $(THOTDIR)/tools/Makefile.in \
-$(THOTDIR)/tools/mkdep/Makefile.in \
-$(THOTDIR)/tools/cextract-1.7/Makefile.in
+$(THOTDIR)/tools/mkdep/Makefile.in
 
 rebuild : tools/mkdep/mkdep Makefile
 
