--- amaya-9.4.orig/Amaya/thotlib/dialogue/AmayaXMLPanel.cpp
+++ amaya-9.4/Amaya/thotlib/dialogue/AmayaXMLPanel.cpp
@@ -102,7 +102,7 @@
   int nb_el                 = (int)p.param1;
   const char * listBuffer   = (char *)p.param2;
   const char * currentEl    = (char *)p.param3;
-  int ref                   = (int)p.param4;;
+  intptr_t ref              = (intptr_t)p.param4;;
   
   m_XMLRef = ref;
   
--- amaya-9.4.orig/Amaya/thotlib/dialogue/printmenu.c
+++ amaya-9.4/Amaya/thotlib/dialogue/printmenu.c
@@ -359,7 +359,7 @@
 #ifndef _WX
        sprintf (&cmd[i], " -w%lu", (intptr_t) FrRef[0]);
 #else /* _WX */
-       sprintf (&cmd[i], " -w%u", (unsigned int) FrameTable[0].WdFrame);
+       sprintf (&cmd[i], " -w%u", (intptr_t) FrameTable[0].WdFrame);
 #endif /* _WX */
 #endif /* _WINDOWS */
      }
--- amaya-9.4.orig/Amaya/thotlib/dialogue/AmayaApplyClassPanel.cpp
+++ amaya-9.4/Amaya/thotlib/dialogue/AmayaApplyClassPanel.cpp
@@ -117,7 +117,7 @@
   int          nb_class = p.param1;
   const char  *listBuffer = (char *)p.param2;
   const char  *currentClass = (char *)p.param3;
-  int          ref = (long int)p.param4;
+  intptr_t          ref = (intptr_t)p.param4;
   
   m_ApplyClassRef = ref;
   
--- amaya-9.4.orig/Amaya/annotlib/ANNOTmenu.c
+++ Amaya-9.4/maya/annotlib/ANNOTmenu.c
@@ -777,7 +777,7 @@
   ----------------------------------------------------------------------*/
 static void AnnotFilterCallbackDialog (int ref, int typedata, char * data)
 {
-  int val;
+  intptr_t val;
 
   if (ref == -1)
     {
@@ -787,7 +787,7 @@
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - AnnotFilterBase)
         {
         case AnnotFilterMenu:
--- amaya-9.4.orig/Amaya/amaya/MENUconf.c
+++ amaya-9.4/Amaya/amaya/MENUconf.c
@@ -4644,14 +4644,14 @@
 static void PreferenceCallbackDialog (int ref, int typedata, char *data)
 {
 #ifdef _WX
-  int val;
+  intptr_t val;
 
   if (ref == -1)
     TtaDestroyDialogue (PreferenceBase);
   else
     {
       /* has the user changed the options? */
-      val = (int) data;
+      val = (intptr_t) data;
       switch (ref - PreferenceBase)
         {
         case 0:
