Index: amaya-9.54~dfsg.0/Makefile.in
===================================================================
--- amaya-9.54~dfsg.0.orig/Makefile.in	2007-02-14 19:09:52.000000000 +0000
+++ amaya-9.54~dfsg.0/Makefile.in	2007-02-14 19:09:59.000000000 +0000
@@ -15,7 +15,7 @@
 
 #MKP: adding davlib
 SUBDIRS	= @SUBDIRS@ tools thotlib batch drawlib indexlib annotlib amaya thot @DAVDIR@
-EXTRA_SUBDIRS= misc redland $(LIBWWW) $(WX_BUILDDIR) $(GL_BUILDDIR)
+EXTRA_SUBDIRS= misc redland raptor $(LIBWWW) $(WX_BUILDDIR) $(GL_BUILDDIR)
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
@@ -24,7 +24,7 @@
 
 srcdir	= $(THOTDIR)
 
-default : tools gl @BUILDWXWIDGETS@ @BUILDREDLAND@ thotlib batch amaya_prog print_prog
+default : tools gl @BUILDWXWIDGETS@ @BUILDREDLAND@ @BUILDRAPTOR@ thotlib batch amaya_prog print_prog
 
 amaya : amaya_prog tools thotlib print_prog
 
@@ -290,6 +290,13 @@
 		@($(ECHO) "libwww is installed")
 
 #
+# Rules to build the Raptor RDF library
+#
+raptor: force
+	$(MAKE) -C raptor
+	@($(ECHO) "raptor is installed")
+
+#
 # Rules to build the redland RDF library
 #
 redland: force
Index: amaya-9.54~dfsg.0/configure.in
===================================================================
--- amaya-9.54~dfsg.0.orig/configure.in	2007-02-14 19:09:52.000000000 +0000
+++ amaya-9.54~dfsg.0/configure.in	2007-02-14 19:09:59.000000000 +0000
@@ -1038,13 +1038,20 @@
    with_bookmarks="no"
 fi
 
-#annot and bookmarks need Redland
-if test "$with_annot" = "yes" -o "$with_bookmarks" = "yes" ; then
+#bookmarks need Redland
+if test "$with_bookmarks" = "yes" ; then
 	with_redland="yes"
 else
 	with_redland="no"
 fi
 
+#annot need Raptor
+if test "$with_annot" = "yes" ; then
+	with_raptor="yes"
+else
+	with_raptor="no"
+fi
+
 AC_ARG_ENABLE([system-redland],
 	AC_HELP_STRING([--enable-system-redland],[try the redland RDF system library]),
 	[case "${enableval}" in
@@ -1054,6 +1061,27 @@
 	esac],
 	[with_system_redland="no"])
 
+AC_ARG_ENABLE([system-raptor],
+	AC_HELP_STRING([--enable-system-raptor],[try the Raptor RDF system library]),
+	[case "${enableval}" in
+	  yes) with_system_raptor="yes";;
+	  no)  with_system_raptor="no" ;;
+	  *) AC_MSG_ERROR(bad value ${enableval} for --enable-system-raptor) ;;
+	esac],
+	[with_system_raptor="no"])
+
+#If Raptor is needed and we are asked
+#to use the system library, check if is present
+if test "$with_raptor" = "yes" -a "$with_system_raptor" = "yes" ; then
+    # check to see if we have a raptor library
+    AC_PATH_PROG([RAPTOR_CONFIG],[raptor-config])
+
+    if test "$RAPTOR_CONFIG" = "" ; then
+	AC_MSG_WARN(Not using system-provided Raptor library)
+	with_system_raptor="no"
+    fi
+fi
+
 #If Redland is needed and we are asked
 #to use the system library, check if is present
 if test "$with_redland" = "yes" -a "$with_system_redland" = "yes" ; then
@@ -1103,18 +1131,32 @@
 	with_redland="no"
 fi
 
+#if we use the builtin library,
+#check it is present
+if test "$with_raptor" = "yes" -a "$with_system_raptor" != "yes" \
+	-a ! -d "$srcdir/../redland/raptor" ; then
+	AC_MSG_WARN(Raptor source dir not found !!)
+	AC_MSG_WARN(Disabling raptor build !)
+	with_raptor="no"
+fi
+
+CURRENTPATH="`pwd`"
+
 #At this point, we know everything about Redland,
 #so either set the variables we will need later,
 #ot disable Annotations and Bookmarks
 if test "$with_redland" = "yes" ; then
-	AMAYA_ANNOT_OPTIONS="$AMAYA_ANNOT_OPTIONS -DAM_REDLAND"
 	if test "$with_system_redland" != "yes" ; then
-		AMAYA_ANNOT_OPTIONS="$AMAYA_ANNOT_OPTIONS -I\$(THOTDIR)/../redland/librdf -I\$(THOTDIR)/../redland/raptor "
-		AMAYA_RAPTOR_LIBS="../redland/librdf/.libs/librdf.a ../redland/raptor/.libs/libraptor.a"
-		AMAYA_RAPTOR_INCLUDES="-I\$(THOTDIR)/../redland/librdf -I\$(THOTDIR)/../redland/raptor"
+
+		if test "$with_system_raptor" = "yes" ; then
+			WITH_RAPTOR="auto"
+		else
+			WITH_RAPTOR="internal"
+		fi
+
 		BUILDREDLAND="redland"
-		CURRENTPATH="`pwd`"
 		mkdir $BUILDREDLAND
+
 		cd $BUILDREDLAND &&
 			CPPFLAGS="-I$CURRENTPATH/../../libwww/modules/expat/lib" \
 			LDFLAGS="-L$CURRENTPATH/libwww/modules/expat -L$CURRENTPATH/libwww/modules/expat/.libs" \
@@ -1123,7 +1165,7 @@
 			--disable-shared \
 			--without-bdb \
 	 		--enable-parsers=raptor \
-	 		--with-raptor=internal \
+	 		--with-raptor=$WITH_RAPTOR \
 			--with-xml-parser=expat \
 			--with-www=none \
 			--without-libwww \
@@ -1131,14 +1173,56 @@
 			--without-mysql
 			$ECHO "redland is configured"
 		cd ..
+
+		REDLAND_CFLAGS="-I\$(THOTDIR)/../redland/librdf"
+		REDLAND_LIBS="../redland/librdf/.libs/librdf.a"
+
+		RAPTOR_CFLAGS="-I\$(THOTDIR)/../redland/raptor"
+		RAPTOR_LIBS="../redland/raptor/.libs/libraptor.a"
 	else
-		redland_includes=`$REDLAND_CONFIG --cflags`
-	  redland_libs=`$REDLAND_CONFIG --libs`
- 		AMAYA_ANNOT_OPTIONS="$AMAYA_ANNOT_OPTIONS"
-		AMAYA_ANNOT_INCLUDES="$redland_includes -I\$(THOTDIR)/thotlib/include"
-		AMAYA_ANNOT_EXTRA_LIBS="$redland_libs"
-		AMAYA_RAPTOR_INCLUDES="$redland_includes"
+		REDLAND_CFLAGS="`$REDLAND_CONFIG --cflags`"
+		REDLAND_LIBS="-lrdf"
+
+		RAPTOR_LIBS="-lraptor"
+
        fi
+elif test "$with_raptor" = "yes" ; then
+	if test "$with_system_raptor" != "yes" ; then
+
+		echo "Configuring Raptor"
+		BUILDRAPTOR="raptor"
+		mkdir $BUILDRAPTOR
+
+		cd $BUILDRAPTOR &&
+			CPPFLAGS="-I$CURRENTPATH/../../libwww/modules/expat/lib" \
+			LDFLAGS="-L$CURRENTPATH/libwww/modules/expat -L$CURRENTPATH/libwww/modules/expat/.libs" \
+			../../../redland/raptor/configure \
+			--build=$build_alias --host=$host_alias --target=$target_alias \
+			--disable-shared \
+			--without-bdb \
+	 		--enable-parsers=raptor \
+	 		--with-raptor=$WITH_RAPTOR \
+			--with-xml-parser=expat \
+			--with-www=none \
+			--without-libwww \
+			--without-openssl-digests \
+			--without-mysql
+			$ECHO "Raptor is configured"
+			$ECHO "Patching the raptor_config.h file to avoid including glib2-0"
+			$CP raptor_config.h raptor_config.h.orig
+			$SED "s/\(\#define HAVE_G_UTF8_NORMALIZE 1\)/\/* \1 *\//" \
+				raptor_config.h.orig > raptor_config.h
+			$RM raptor_config.h.orig
+			$ECHO "Raptor is patched"
+		cd ..
+
+		RAPTOR_CFLAGS="-I\$(THOTDIR)/../redland/raptor"
+		RAPTOR_LIBS="../raptor/.libs/libraptor.a"
+	else
+		RAPTOR_CFLAGS="`$RAPTOR_CONFIG --cflags`"
+		RAPTOR_LIBS="-lraptor"
+	fi
+	with_bookmarks="no"
 else
 	with_annot="no"
 	with_bookmarks="no"
@@ -1147,12 +1231,30 @@
 #Bookmarks require some specific flags
 if test "$with_bookmarks" = "yes" ; then
 	ANNOTLIB_COMPILE_BM=
-	AMAYA_ANNOT_OPTIONS="$AMAYA_ANNOT_OPTIONS -DBOOKMARKS"
+	ANNOTATIONS_CFLAGS="-DBOOKMARKS"
 else
 	ANNOTLIB_COMPILE_BM=['#']
 fi
 
+if test "$with_annot" = "yes" ; then
+	ANNOTATIONS_CFLAGS="$ANNOTATIONS_CFLAGS -DANNOTATIONS"
+	ANNOTATIONS_INCLUDES="-I\$(THOTDIR)/annotlib -I\$(THOTDIR)/annotlib/f"
+	ANNOTATIONS_LIBS="../annotlib/libAnnot.a"
+fi
+
+AC_SUBST(ANNOTATIONS_CFLAGS)
+AC_SUBST(ANNOTATIONS_INCLUDES)
+AC_SUBST(ANNOTATIONS_LIBS)
+
+AC_SUBST(ANNOTLIB_COMPILE_BM)
+
 AC_SUBST(BUILDREDLAND)
+AC_SUBST(REDLAND_CFLAGS)
+AC_SUBST(REDLAND_LIBS)
+
+AC_SUBST(BUILDRAPTOR)
+AC_SUBST(RAPTOR_CFLAGS)
+AC_SUBST(RAPTOR_LIBS)
 
 ###################End of Annotations and Bookmarks Support####################
 
@@ -1357,9 +1459,6 @@
 AC_SUBST(GTK_OPTIONS)
 AC_SUBST(THOT_OPTIONS)
 AC_SUBST(AMAYA_OPTIONS)
-AC_SUBST(AMAYA_ANNOT_OPTIONS)
-AC_SUBST(AMAYA_ANNOT_INCLUDES)
-AC_SUBST(AMAYA_ANNOT_EXTRA_LIBS)
 AC_SUBST(IMGLIBS)
 AC_SUBST(EXTRA_INCLUDES)
 
@@ -1368,9 +1467,6 @@
 AC_SUBST(WITHDAV)
 AC_SUBST(WWWDAV)
 
-AC_SUBST(ANNOTLIB_COMPILE_BM)
-AC_SUBST(AMAYA_RAPTOR_LIBS)
-AC_SUBST(AMAYA_RAPTOR_INCLUDES)
 AC_SUBST(MAKE_LIBWWW_RDF_PARSER)
 AC_SUBST(LIBWWW_RDF_PARSER)
 
Index: amaya-9.54~dfsg.0/annotlib/Makefile.annot
===================================================================
--- amaya-9.54~dfsg.0.orig/annotlib/Makefile.annot	2007-01-22 11:42:34.000000000 +0000
+++ amaya-9.54~dfsg.0/annotlib/Makefile.annot	2007-02-14 19:09:59.000000000 +0000
@@ -4,22 +4,10 @@
 # Jose Kahan, Sep 1999
 #
 
-AMAYA_ANNOT_OPTIONS+= -DANNOTATIONS
-
-AMAYA_ANNOT_INCLUDES+= -I$(THOTDIR)/annotlib -I$(THOTDIR)/annotlib/f
-
-AMAYA_ANNOT_OBJ=
-
 AMAYA_ANNOT_SRC= ../annotlib/libAnnot.a 
 
-AMAYA_ANNOT_LIBS= ../annotlib/libAnnot.a
-
 ANNOT_SCHEMAS= annot_schemas
 
-AMAYA_ANNOT_INSTALL=
-
-AMAYA_ANNOT_UNINSTALL=
-
 annot_force :
 
 amaya_annot_schemas : annot_force
Index: amaya-9.54~dfsg.0/annotlib/Makefile.in
===================================================================
--- amaya-9.54~dfsg.0.orig/annotlib/Makefile.in	2007-01-22 11:42:34.000000000 +0000
+++ amaya-9.54~dfsg.0/annotlib/Makefile.in	2007-02-14 19:09:59.000000000 +0000
@@ -10,17 +10,23 @@
 THOTDIR= @top_srcdir@
 ANNOTLIB= $(THOTDIR)/annotlib
 
+RAPTOR_CFLAGS=@RAPTOR_CFLAGS@
+REDLAND_CFLAGS=@REDLAND_CFLAGS@
+
+ANNOTATIONS_CFLAGS = @ANNOTATIONS_CFLAGS@
+
 include ../Options
 
-INCLUDES= -DHAVE_CONFIG_H -DANNOTATIONS \
+INCLUDES= -DHAVE_CONFIG_H \
 	  -I.. -I../amaya  -I@srcdir@/f -I@top_srcdir@/amaya \
 	  -I@top_srcdir@/amaya/f -I../libwww \
 	  -I@top_srcdir@/../libwww/Library/src \
 	  -I@top_srcdir@/../libwww/modules/expat/lib \
-	   @AMAYA_RAPTOR_INCLUDES@ @THOTINCLUDES@ $(GUI_INCLUDES)
+	   $(RAPTOR_CFLAGS) $(REDLAND_CFLAGS) @THOTINCLUDES@ $(GUI_INCLUDES)
+
 
+OPTIONS	= @AMAYA_OPTIONS@ $(ANNOTATIONS_CFLAGS)
 
-OPTIONS	= @AMAYA_OPTIONS@ @AMAYA_ANNOT_OPTIONS@
 
 prefix = @prefix@
 exec_prefix = @exec_prefix@
Index: amaya-9.54~dfsg.0/amaya/Makefile.in
===================================================================
--- amaya-9.54~dfsg.0.orig/amaya/Makefile.in	2007-02-14 19:09:52.000000000 +0000
+++ amaya-9.54~dfsg.0/amaya/Makefile.in	2007-02-14 19:09:59.000000000 +0000
@@ -12,11 +12,13 @@
 
 OBJDIRS = wxdialog
 
+RAPTOR_LIBS=@RAPTOR_LIBS@
+REDLAND_LIBS=@REDLAND_LIBS@
+
 include ../Options
 
 INCLUDES= -DHAVE_CONFIG_H $(AMAYA_INCLUDES) $(EXPAT_INCLUDES) $(GUI_INCLUDES) $(EXTRA_INCLUDES) $(GL_INCLUDES) $(GTK_GL_INCLUDES) $(FREETYPE_INCLUDES) $(XFT_INCLUDES)
-LIBS	=  $(AMAYA_ANNOT_LIBS) $(AMAYA_DAV_LIBS) \
-	   @AMAYA_RAPTOR_LIBS@ \
+LIBS	=  $(AMAYA_DAV_LIBS) \
 	   $(AMAYA_MATH_LIBS) $(AMAYA_GRAPH_LIBS) \
            -L../thotlib -L.. -lThotEditor \
           $(AMAYA_OPTION_EXTRA_LIBS) $(IMGLIBS) $(EXPAT_LIBRARIES) \
@@ -132,16 +134,16 @@
 ALL_AMAYA_OPTIONS= $(AMAYA_OPTIONS) \
 	 $(AMAYA_LIBWWW_OPTIONS) \
 	 $(AMAYA_MATH_OPTIONS) $(AMAYA_SVG_OPTIONS)  $(AMAYA_TEMPLATES_OPTIONS) \
-	 $(AMAYA_ANNOT_OPTIONS) $(AMAYA_DAV_OPTIONS) $(EXPAT_OPTIONS)
+	 $(ANNOTATIONS_CFLAGS) $(AMAYA_DAV_OPTIONS) $(EXPAT_OPTIONS)
 
 AMAYA_OPTION_INCLUDES= \
          $(AMAYA_LIBWWW_INCLUDES) $(AMAYA_MATH_INCLUDES) \
-	 $(AMAYA_SVG_INCLUDES) $(AMAYA_ANNOT_INCLUDES) \
+	 $(AMAYA_SVG_INCLUDES) $(ANNOTATIONS_INCLUDES) \
 	 $(AMAYA_DAV_INCLUDES) $(AMAYA_TEMPLATES_INCLUDES)
 
 AMAYA_OPTION_OBJ= \
          $(AMAYA_LIBWWW_OBJ) $(AMAYA_SVG_OBJ) $(AMAYA_MATH_OBJ) \
-         $(AMAYA_ANNOT_OBJ) $(AMAYA_DAV_OBJ) $(AMAYA_TEMPLATES_OBJ)
+         $(AMAYA_DAV_OBJ) $(AMAYA_TEMPLATES_OBJ)
 
 AMAYA_SRC= \
          $(AMAYA_LIBWWW_SRC) @BUILDREDLAND@ \
@@ -151,11 +153,11 @@
 
 AMAYA_OPTION_LIBS= \
          $(AMAYA_LIBWWW_OPTION_LIBS) $(AMAYA_MATH_LIBS) $(AMAYA_GRAPH_LIBS) \
-         $(AMAYA_ANNOT_LIBS) $(AMAYA_DAV_LIBS) @AMAYA_RAPTOR_LIBS@
+         $(ANNOTATIONS_LIBS) $(AMAYA_DAV_LIBS)
 
 AMAYA_OPTION_EXTRA_LIBS= \
          $(AMAYA_LIBWWW_EXTRA_LIBS) $(AMAYA_MATH_EXTRA_LIBS) \
-         $(AMAYA_GRAPH_EXTRA_LIBS) $(AMAYA_ANNOT_EXTRA_LIBS)
+         $(AMAYA_GRAPH_EXTRA_LIBS) $(REDLAND_LIBS) $(RAPTOR_LIBS)
 
 AMAYA_OPTION_SCHEMAS= \
          $(AMAYA_LIBWWW_SCHEMAS) $(MATH_SCHEMAS) $(SVG_SCHEMAS) \
Index: amaya-9.54~dfsg.0/davlib/Makefile.in
===================================================================
--- amaya-9.54~dfsg.0.orig/davlib/Makefile.in	2007-01-22 11:42:34.000000000 +0000
+++ amaya-9.54~dfsg.0/davlib/Makefile.in	2007-02-14 19:09:59.000000000 +0000
@@ -87,7 +87,7 @@
 # C files
 #
 #.c.o :  
-#	$(CC) $(CFLAGS) $(AMAYA_ANNOT_OPTIONS) $(INCLUDES) $(OPTIONS) -c $< -o $@
+#	$(CC) $(CFLAGS) $(INCLUDES) $(OPTIONS) -c $< -o $@
 
 
 
